### PR TITLE
[CE-919] Add AWS Codebuild support

### PR DESCRIPTION
**Pull request by drazisil**
## Purpose
Add AWS Codebuild support

## Notable Changes
No

## Tests and Risks?
New tests

### CODE REVIEW

**Comment by drazisil** [comment]
Closes #216

File: tests/env [hunk 1]
```diff
@@ -110,3 +110,10 @@ export BITBUCKET_REPO_OWNER=""
 export BITBUCKET_REPO_SLUG=""
 export BITBUCKET_PR_ID=""
 export GITHUB_ACTION=""
+export CODEBUILD_BUILD_ARN=""
+export CODEBUILD_WEBHOOK_HEAD_REF=""
+export CODEBUILD_RESOLVED_SOURCE_VERSION=""
+export CODEBUILD_SOURCE_VERSION=""
+export CODEBUILD_BUILD_ID=""
+export CODEBUILD_SOURCE_REPO_URL=""
+export CODEBUILD_BUILD_ID=""
```
Comment (ibrahim0814) [inline tests/env]: Why is  `CODEBUILD_BUILD_ID=""` on this list twice?

**Comment by ibrahim0814** [comment]
This is a good PR, shoutout to you @drazisil for killing it on adding CI support

File: codecov [hunk 2]
```diff
@@ -492,6 +492,22 @@ then
     env="$env,${!language}"
   fi
 
+elif [ "$CODEBUILD_BUILD_ARN" != "" ];
+then
+  say "$e==>$x AWS Codebuild detected."
+  # https://docs.aws.amazon.com/codebuild/latest/userguide/build-env-ref-env-vars.html
```
Comment (drazisil) [inline codecov]: Right here, @hootener :)

**Comment by drazisil** [comment]
This has been deployed

**Pull request merged**